### ISSUE TITLE
Make possible to handle connection and socket errors

### ISSUE THREAD

**Issue by ribizli** [hbenl/mpc-js-node#1]
```
events.js:163
      throw er; // Unhandled 'error' event
      ^

Error: connect ECONNREFUSED 127.0.0.1:6600
```
I think this needs a bit restructuring, since for the outside world only the MPC events are visible. Maybe the SocketWrapper.connect method should have a second `onError` callback which callback forwards and emits the error from MPC.

**Comment by hbenl** [comment]
The `error` and `end` events from the socket are now forwarded by MPC as `socket-error` and `socket-end` events. The `socket-end` event signals that the server (i.e. mpd) has closed the connection - this may be useful because it is fired immediately when the connection is closed, whereas the `error` event will only be fired when you try to send a message to mpd after the connection was closed.

Also, the `connectTCP` method (and the other `connect` methods) now return a `Promise` that is resolved when the connection is established and rejected when the connection attempt fails. This is easier than the `socket-error` event for handling failed connection attempts, but you will still need the `socket-error` and/or `socket-end` events to handle the connection failing *after* it was established.

> Maybe the SocketWrapper.connect method should have a second onError callback which callback forwards and emits the error from MPC.

Yup, that's exactly what I did :)
